### PR TITLE
[MNT] Type hint distance module

### PR DESCRIPTION
**Pull request by chrisholder**
#### What does this implement/fix? Explain your changes.

This PR adds correct type hinting to the entire distance module.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [x] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed.

##### For developers with write access
- [x] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

**Comment by chrisholder** [comment]
Depends on PR #1326

**Comment by MatthewMiddlehurst** [comment]
Yes this should wait until a decision is made there. The failures seem unrelated however, as the dependency is already installed via (I think) the developer install packages.

**Review by TonyBagnall — DISMISSED** [review]
blimey, that must have been boring :)

**Comment by chrisholder** [comment]
Will hold merging this until the typing-extension PR has gone through as it uses a feature of that dependency (anyone who just is running aeon with minimum dependency will get errors)

File: aeon/distances/_distance.py [hunk 1]
```diff
@@ -78,6 +79,26 @@
 )
 from aeon.distances.mpdist import mpdist
 
+
+class DistanceKwargs(TypedDict, total=False):
+    window: float
+    itakura_max_slope: None
```
[the remark this answers is not in the document]
Comment (chrisholder) [reply aeon/distances/_distance.py:85]: Good spot!

**Comment by chrisholder** [comment]
Since a previous update by @CodeLionX fixed how univariate arrays were being converted in pairwise computation the examples broke. I'm not sure why the CI didn't pick it up in his PR (think it's the numba caching), but I have also fixed them in this PR.

**Pull request merged**